### ISSUE TITLE
`nn.functional.fractional_max_pool3d` produces incorrect output on non-contiguous inputs

### ISSUE THREAD

**Issue by saketh-are** [pytorch/pytorch#69325]
To reproduce, disable the corresponding expectedFailure in its OpInfo:

https://github.com/pytorch/pytorch/pull/68922/files/aef50820ab2048e09f31d6a948ed46d0bd44ed80#diff-18a0c3e7cd07089b34760028c387d39de1aa393172f1d5f123692106b3a3d144R10451

**Comment by XiaobingSuper** [comment]
@saketh-are , I don't think it is an issue, because **fractional_max_pool2d** ouptut depend on **_random_samples** , but your sample inputs do not have it, so the output of **fractional_max_pool2d** is always random if *_random_samples** is not gven,  see
https://github.com/pytorch/pytorch/blob/48771d1c7f01e5b26c5bf0d6d8973884f908453f/torch/nn/functional.py#L470-L472.
